dev/com.ibm.ws.microprofile.openapi.ui/src/com/ibm/ws/microprofile/openapi/ui/OpenAPIUIEndpointManager.java

```diff
@@ -0,0 +1,153 @@
+
+    private static final String OPEN_API_UI_VAR_NAME = "openAPIUIPATH";
+    private static final String OPEN_API_UI_CONFIG_NAME = "mp.openapi.extensions.liberty.ui.path";
+    private static final String OPEN_API_UI_PATH = "/ui";
+
+    private static final Pattern PATH_PATTERN = Pattern.compile("^(/[\\w./_-]*)?$");
+    private static final Map<String, String> CONFLICTING_PATHS_MAP = Stream.of(
+                                                                               new AbstractMap.SimpleEntry<>("/ibm/api", "OpenAPI"),
+                                                                               new AbstractMap.SimpleEntry<>("/health", "mpHealth"),
+                                                                               new AbstractMap.SimpleEntry<>("/metrics", "mpMetrics")).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    private String uiPath = null;
+    private String dataPath = null;
+
+    private OpenAPIWABConfigManager openAPIWABConfigManager;
+
+    @Activate
+    protected void activate(ComponentContext context, Map<String, Object> properties) {
+        //Default UI Path is /openapi/ui
+        uiPath = OPEN_API_ENDPOINT_PATH + OPEN_API_UI_PATH;
+
+        if (ProductInfo.getBetaEdition()) {
+            Config config = ConfigProvider.getConfig(OpenAPIUIEndpointManager.class.getClassLoader());
+            if (config.getOptionalValue("open_api_path_enabled", Boolean.class).orElse(false)) {
+                resolvePathFromConfig(config);
+                if (!validatePath(uiPath, dataPath)) {
+                    uiPath = dataPath + OPEN_API_UI_PATH;
+                    Tr.warning(tc,"CWWKO1751W_OPEN_API_PATH_UPDATE_FAILED");
+                } else if(!uiPath.equals(OPEN_API_ENDPOINT_PATH + OPEN_API_UI_PATH)){
+                    Tr.info(tc,"CWWKO1750I_OPEN_API_PATH_UPDATE",uiPath);
+                }
+            }
+        }
+        openAPIWABConfigManager = new OpenAPIWABConfigManager(context, OPEN_API_UI_VAR_NAME, uiPath, "OpenAPI UI");
+        openAPIWABConfigManager.activate();
+    }
+
+    @Deactivate
+    protected void deactivate(ComponentContext context, int reason) {
+        openAPIWABConfigManager.deactivate();
+    }
+
+    /**
+     * Validate the OpenAPI path to ensure it is Valid, contains only valid characters and mets certain rules.
+     *
+     * Reserved Path is a related entitiy where the path and the reserved path cannot be the same*
+     *
+     * @param path
+     * @param reservedPath
+     * @return
+     */
+    public static boolean validatePath(String path, String reservedPath) {
+        boolean valid = true;
+        try {
+            //temporarily construct a URL using provided path to use the URL validation code.
+            new URL("https://localhost" + path);
+        } catch (MalformedURLException e) {
+            Tr.warning(tc, "CWWKO1754W_OPEN_API_UI_PATH_INVALID",e.getLocalizedMessage());
+            return false;
+        }
```

Comment:
Not convinced this is needed. In addition, the message produced here is likely to be less useful than those produced by the following lines.

We'll address it under another issue.